pkg/virt-handler/vm.go

```diff
@@ -1904,18 +1904,40 @@ func (d *VirtualMachineController) isPreMigrationTarget(vmi *v1.VirtualMachineIn
 }
 
 func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.VirtualMachineInstance) error {
-	networks := map[string]*v1.Network{}
-	for _, network := range vmi.Spec.Networks {
-		networks[network.Name] = network.DeepCopy()
+	err := validatePodNetworkInterfaceUsesMasqueradeBinding(vmi)
+	if err != nil {
+		return err
 	}
+
+	return nil
+}
+
+func validatePodNetworkInterfaceUsesMasqueradeBinding(vmi *v1.VirtualMachineInstance) error {
+	podNetwork := getVMIPodNetwork(vmi)
+
+	interfacesLookup := map[string]v1.Interface{}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Masquerade == nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
-		}
+		interfacesLookup[iface.Name] = iface
 	}
+
+	if podNetwork != nil && interfacesLookup[podNetwork.Name].Masquerade == nil {
+		return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
+	}
+
 	return nil
 }
 
+func getVMIPodNetwork(vmi *v1.VirtualMachineInstance) *v1.Network {
+	var podNetwork *v1.Network
+	for _, network := range vmi.Spec.Networks {
+		if network.Pod != nil {
+			podNetwork = &network
```

Comment:
It is unsafe to return the for-value address in general.
But you do not need all this anyway, just return the name or a the whole network as a value.
(if it still fitting to return the network itself, use the index (`return &vmi.Spec.Networks[i]`)

Note: You do not need the `podNetwork` variable as well, it is clearer that you return nil or an empty string if it was not found.

Reply:
I will change it to return the name it is sufficient for now.

Out of curiosity why is it unsafe to return the for-value address?

About the note, I find function that have single return more readable and much easier to follow
what you think?